backend/src/main/kotlin/com/pk/zbtz/zbtzbackend/databases/postgres/services/MovieService.kt

```diff
@@ -0,0 +1,106 @@
+
+import com.pk.zbtz.zbtzbackend.com.pk.zbtz.zbtzbackend.databases.postgres.models.ActorPostgresModel
+import com.pk.zbtz.zbtzbackend.com.pk.zbtz.zbtzbackend.databases.postgres.models.DirectorPostgresModel
+import com.pk.zbtz.zbtzbackend.com.pk.zbtz.zbtzbackend.databases.postgres.models.HumanMovieId
+import com.pk.zbtz.zbtzbackend.com.pk.zbtz.zbtzbackend.databases.postgres.repositories.GenresPostgresRepository
+import com.pk.zbtz.zbtzbackend.com.pk.zbtz.zbtzbackend.databases.postgres.repositories.PlatformsPostgresRepository
+import com.pk.zbtz.zbtzbackend.controllers.movie.requests_and_responses.AddMovieRequest
+import com.pk.zbtz.zbtzbackend.databases.postgres.models.MoviePostgresModel
+import com.pk.zbtz.zbtzbackend.databases.postgres.repositories.MoviePostgresRepository
+import org.springframework.stereotype.Service
+import java.util.UUID
+
+@Service
+class MovieService(
+    private val movieRepository: MoviePostgresRepository,
+    private val platformsPostgresRepository: PlatformsPostgresRepository,
+    private val genresPostgresRepository: GenresPostgresRepository,
+    private val humanService: HumanService,
+    private val actorService: ActorService,
+    private val directorService: DirectorService
+) {
+    fun getById(id: Int): MoviePostgresModel? {
+        return movieRepository.findById(id).orElseThrow()
+    }
+
+    fun getAll(offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllWithOffset(offset)
+    }
+
+    fun getAllByYear(year: Int, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByYearWithOffset(year, offset)
+    }
+
+    fun getAllByPlatform(platform: String, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByPlatformWithOffset(platform, offset)
+    }
+
+    fun getAllByTitle(title: String, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByTitleWithOffset("%$title%", offset)
+    }
+
+    fun getAllByTitleAndPlatform(title: String, platform: String, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByTitleAndPlatformWithOffset("%$title%", platform, offset)
+    }
+
+    fun getAllByTitleAndYear(title: String, year: Int, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByTitleAndYearWithOffset("%$title%", year, offset)
+    }
+
+    fun getAllByPlatformAndYear(platform: String, year: Int, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByPlatformAndYearWithOffset(platform, year, offset)
+    }
+
+    fun getAllByPlatformAndTitleAndYear(platform: String, title: String, year: Int, offset: Int): List<MoviePostgresModel> {
+        return movieRepository.findAllByPlatformAndYearAndTitleWithOffset(platform, year, "%$title%", offset)
+    }
+
+    fun save(request: AddMovieRequest): MoviePostgresModel {
+        val id = UUID.randomUUID()
```

Comment:
I think we don't need this id, because later is set to `null`
```suggestion
```

Reply:
yep